healthz/clusterstatus/clusterstatus.go

```diff
@@ -0,0 +1,104 @@
+package clusterstatus
+
+import (
+	"bufio"
+	"bytes"
+	"fmt"
+	"strings"
+	"sync"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/gravitational/satellite/healthz/config"
+	"github.com/gravitational/satellite/healthz/utils"
+)
+
+type ClusterStatus struct {
+	K8S struct {
+		Status       bool
+		NodesTotal   uint
+		NodesHealthy uint
+	}
+	Etcd struct {
+		Status bool
+	}
+	Mux sync.Mutex
+}
+
+const ContextKey = "clusterstatus"
+
+func (c *ClusterStatus) String() string {
+	c.Mux.Lock()
+	defer c.Mux.Unlock()
+	return fmt.Sprintf("{\"K8S\": {\"Status\": %v, \"NodesTotal\": %v "+
+		"\"NodesHealthy\": %v}, \"Etcd\": {\"Status\": %v}}", c.K8S.Status,
+		c.K8S.NodesTotal, c.K8S.NodesHealthy, c.Etcd.Status)
+}
+
+func (c *ClusterStatus) updateK8sClusterStatus() {
+	_, err := utils.ExecCommand("kubectl", "cluster-info")
+	c.Mux.Lock()
+	defer c.Mux.Unlock()
+	c.K8S.Status = (err == nil) // No errors -- healthy
+}
+
+func (c *ClusterStatus) updateEtcdStatus() {
+	_, err := utils.ExecCommand("etcdctl", "cluster-health")
```

Comment:
same here, instead of relying on the binary, please use HTTP client, e.g. like one here :

https://github.com/gravitational/planet/blob/master/lib/monitoring/checkers.go#L78